### PR TITLE
Extract Offerings-related things into OfferingsAPI

### PR DESCRIPTION
**Pull request by taquitos**
Resolves [CSDK-154]
Extracts all the remaining logic from `Backend.swift` leaving only proxied methods.

[CSDK-154]: https://revenuecats.atlassian.net/browse/CSDK-154?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by taquitos** [comment]
Note: I also reorganized the `Networking` folder so that all the HTTPClient-related things went into a HTTPClient group, leaving only the `Backend`, `BackendConfiguration`, and API classes.

File: Sources/Networking/Backend.swift [hunk 1]
```diff
@@ -181,3 +149,7 @@ extension Backend {
     }
 
 }
+
+extension Backend {
+
+}
```
Comment (NachoSoto) [inline Sources/Networking/Backend.swift:155]: Looks like this is a leftover?

**Review by NachoSoto — APPROVED** [review]
😻

File: Sources/Networking/Backend.swift (hunk 1 above)
Comment (taquitos) [reply Sources/Networking/Backend.swift]: ✅

**Pull request merged**